client/client.go

```diff
@@ -393,12 +393,44 @@ func (c *Client) Subscribe(topic string, qos uint8, callback MessageHandlerFunc)
 	}
 }
 
+func (c *Client) unsubscribe(topicName string, topicIDType uint8, topicID uint16) error {
```

Comment:
You've placed `unsubscribe()` and `Unsubscribe()` between `Subscribe()` and `SubscribePredefinedTopic()`. I think we should keep all subscribe-related functions together and all unsubscribe-related functions together.